cmd/krew/cmd/update.go

```diff
@@ -102,13 +104,24 @@ func showUpdatedPlugins(out io.Writer, preUpdate, posUpdate []index.Plugin, inst
 	}
 }
 
-func ensureIndexUpdated(_ *cobra.Command, _ []string) error {
+func ensureIndexesUpdated(_ *cobra.Command, _ []string) error {
 	preUpdateIndex, _ := indexscanner.LoadPluginListFromFS(paths.IndexPluginsPath(constants.DefaultIndexName))
 
-	klog.V(1).Infof("Updating the local copy of plugin index (%s)", paths.IndexPath(constants.DefaultIndexName))
-	if err := gitutil.EnsureUpdated(constants.DefaultIndexURI, paths.IndexPath(constants.DefaultIndexName)); err != nil {
-		return errors.Wrap(err, "failed to update the local index")
+	indexes, err := indexoperations.ListIndexes(paths)
+	if err != nil {
+		return errors.Wrap(err, "failed to list indexes")
+	}
+
+	var failed []string
+	for _, idx := range indexes {
+		indexPath := paths.IndexPath(idx.Name)
+		klog.V(1).Infof("Updating the local copy of plugin index (%s)", indexPath)
+		if err := gitutil.EnsureUpdated(idx.URL, indexPath); err != nil {
+			klog.V(1).Infof("error updating index %s: %s", idx.Name, err)
```

Comment:
you should probably do this as `klog.Warningf` otherwise the user will never know (unless they use -v) what the error is.

See how we handle multiple plugin install failures in `install.go`.